File: mobile-client/android/UseCase/System/Billing/VerifyReceiptUseCase.kt
package app.mobiling.client.client.usecase.system.billing

import app.mobiling.client.client.contract.system.billing.ReceiptVerificationResult
import app.mobiling.client.client.data.system.billing.BillingReceiptGateway

class VerifyReceiptUseCase(private val billingReceiptGateway: BillingReceiptGateway) {
    operator fun invoke(token: String): ReceiptVerificationResult =
        billingReceiptGateway.verify(token)
}

File: client/ios/UseCase/System/Billing/VerifyReceiptUseCase.swift
import Foundation

public struct VerifyReceiptUseCase {
    private let billingReceiptGateway: BillingReceiptGateway

    public init(billingReceiptGateway: BillingReceiptGateway) {
        self.billingReceiptGateway = billingReceiptGateway
    }

    public func callAsFunction(token: String) async throws -> ReceiptVerificationResult {
        try await billingReceiptGateway.verify(token: token)
    }
}
